pkg/log/log.go

```diff
@@ -207,3 +210,8 @@ func generateFatalFields(entry F) {
 	entry["error.message"] = "fatal occurred"
 	entry["error.stack"] = string(debug.Stack())
 }
+
+var appenders = make([]func(ctx context.Context) Marshaler, 0)
```

Comment:
I think this is awkward.  Why not just change trace as our default logging package as well (and have it call log with the current trace info added) ?

i.e. we extend trace to include all the default log functions but it also adds the trace info.  

We would need to change all shared library and related code to use trace instead of log but that seems relatively simple.  Or we could just make log.XXX point to trace.XXX.

Reply:
Another approach:

1. move all logging functionality into pkg/logger but don't use globals (instead require log.New).
2. move trace to define a DefaultLogger and switch to using that.  And also extend trace to implement the log.XXX equivalents by automatically adding the trace Info.
3. point log.XXX to use trace.XXX instead.

Reply:
We already polluted trace with trace.Call business (that includes logging) and it might be time to just admit that tracing and logging are too coupled together for us.

Reply:
Yeah. I had various versions but wanted to create minimal change. I see it following;
Current situation:
![image](https://user-images.githubusercontent.com/70577857/150542857-f9f509a5-4520-4263-8da9-7329c74edccf.png)
IMHO too messy, no need to say that in tracing we mix plenty of other stacks (http headers, rountrippers, etc) All this needs its own place.Back to the topic. 

Logging, tracing and metering serve its own purpose. We want to combine some of the aspects (logging+tracing) in some cases so that developers can rely on the fact. that tracing of the calls (btw, same as span just differs by logging/metering aspect) emits log+trace+meter. IMO, metering. can be solved by log-to-metrics so I'd like to skip it for now. 

The topic is too broad here for the PR, I'm preparing a document discussing it but until that, solving tracing ID in the logs with minimal change - what about pushing "trace ID info" to the context and let the logger use the logging context and emit it. **Internal** for now until we solve log context correlation topic.